tests/link_types.cpp

```diff
@@ -0,0 +1,303 @@
+////////////////////////////////////////////////////////////////////////////
```

Comment:
This entire file appears to just be testing core functionality, and should be part of core's test suite instead.